tests/unit/keyinput/test_basekeyparser.py

```diff
@@ -268,3 +348,13 @@ def test_count_42_invalid(self, fake_keyevent_factory):
         self.kp.handle(fake_keyevent_factory(Qt.Key_A, text='c'))
         self.kp.execute.assert_called_once_with('ccc', self.kp.Type.chain, 23)
         assert self.kp._keystring == ''
+
+
+def test_clear_keystring(qtbot):
+    """Test that the keystring is cleared and the signal is emitted"""
+    kp = basekeyparser.BaseKeyParser(0)
+    kp._keystring = 'test'
+    kp.clear_keystring()
+    with qtbot.waitSignal(kp.keystring_updated):
+        pass
```

Comment:
This is one of the cases where the code *generating* the signal should be in the with-block (like my example I made in IRC):

```python
with qtbot.waitSignal(kp.keystring_updated):
    kp.clear_keystring()
```

I think this is much easier to understand - the only reason we can't do that in `test_delayed_exec` is that the `keystring_updated` signal actually gets emitted in the background after the delay.